torch/onnx/symbolic_opset9.py

```diff
@@ -6019,6 +6019,13 @@ def linalg_vector_norm(
                 ord_op,
             ),
         )
+
+        out_dtype = self.type().dtype()
+        if out_dtype == torch.float16:
+            result = g.op("Cast", result, to_i=_C_onnx.TensorProtoDataType.FLOAT16)
+        elif out_dtype == torch.bfloat16:
+            result = g.op("Cast", result, to_i=_C_onnx.TensorProtoDataType.BFLOAT16)
```

Comment:
```suggestion
        out_dtype = _type_utils.JitScalarType.from_value(self)
        if _try_get_scalar_type(k) != _type_utils.JitScalarType.FLOAT:
            result = g.op("Cast", result, to_i=out_dtype.onnx_type())
```
Maybe something like this (not tested)